indigo/views.py

```diff
@@ -982,25 +985,29 @@ def admin_projects_new(request):
         if form.is_valid():
             # process the data in form.cleaned_data as required
             # Save the event
-            id = form.cleaned_data["id"]
-            existing_record = Record.objects.filter(type=type, public_id=id)
-            if existing_record:
-                form.add_error("id", "This ID already exists")
-            else:
-                data = NewEventData(
-                    type,
-                    id,
-                    {"name": {"value": form.cleaned_data["name"]}},
-                    approved=True,
-                )
-                newEvent(
-                    [data], user=request.user, comment=form.cleaned_data["comment"]
+            last_record = (
+                Record.objects.filter(type=type).order_by("-public_id").first()
```

Comment:
doing it on the Record class rather than Project class is good, helps avoid any race issues :+1: